Assets/_Laz/Scripts/Environment/ElectricGate.cs

```diff
@@ -0,0 +1,77 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Laz
+{
+    [RequireComponent(typeof(BoxCollider))]
+    public class ElectricGate : MonoBehaviour
+    {
+
+        [SerializeField, Range(0f, 50f)] float _gateWidth = 10f;
+        [SerializeField] Transform _leftPost, _rightPost, _gate = default;
+        
+
+        private BoxCollider _gateCollider = null;
+
+        private void Awake()
+        {
+            _gateCollider = GetComponent<BoxCollider>();
+            ScaleGate();
```

Comment:
Is scaling actually part of the Acceptance Critera? Would it just be easier for the 2 poles be set at the child of the gate, and then @wbeltran777 can scale the gate?

Reply:
At the end of the day, as long as william is okay with this way of scaling, this is fine.

Reply:
I see what you're saying, and I had originally just left the scaling alone - but I couldn't think of a way to set up the hierarchy so that the poles were both consistently on the edge of the gate and weren't being squished by the scaling. I think the serializedField won't be too annoying, cause it is getting updated in OnValidate and is mostly a proxy for the gate's z scale, so it should be relatively easy to see what's happening in the editor

Reply:
Alright sounds good